CONTRIBUTING.md

```diff
@@ -1,42 +1,229 @@
+   at least two official maintainers.
+8. One of the maintainers will merge the PR into the main repo.
+
+
+### Code Standards
+For a smooth code review, it helps to make sure your code adheres to standards,
+conventions, and design goals for OpenHTF. A best-effort attempt to understand 
+and meet these standards before requesting code review can go a long way towards
+making the review process as fast and painless as possible.
+
+
+#### Follow Style Guidelines
+OpenHTF's Python code follows the
+[Google Python Style Guide](https://google.github.io/styleguide/pyguide.html).
+We provide a `pylintrc` file at the top level of our repo that you can use with
+[pylint](https://www.pylint.org/) to lint your Python code.
+
+OpenHTF's built-in web frontend, written in TypeScript and using Angular2,
+follows the [Angular2 Style Guide](https://angular.io/styleguide).
+We recommend using [codelyzer](https://www.npmjs.com/package/codelyzer) to lint
+your frontend code.
+
+
+#### Provide Documentation
+OpenHTF code is fairly well-documented through the liberal use of docstrings and
+comments in the actual code. If your code is doing something that isn't
+immediately intuitive, consider adding clarifying comments.
+
+In addition to inline documentation, adding example scripts to the `examples/`
+directory is a great way to express the intended usage of OpenHTF features.
+
+
+#### Limit Complexity
+Code that is complex requires programmers to hold many things in their heads
+simultaneously in order to make modifications and fixes. Try to write simple,
+composable components with clean, well-defined interfaces wherever possible.
+Check out
+[this great talk](https://www.infoq.com/presentations/Simple-Made-Easy)
+for a deeper look at code complexity.
+
+
+#### Consider OpenHTF's Goals
+OpenHTF is designed to abstract away nearly all boilerplate of hardware test
+setup and execution, so test engineers can focus entirely on writing actual
+test logic. It aspires to do so in a lightweight and minimalistic fashion. OpenHTF should remain general enough to be useful in a variety of hardware testing scenarios, from the lab bench to the manufacturing floor.
+
+Try to keep these goals in mind when authoring your contributions.
+
+
+#### Stick to Existing Structure
+The package/module structure of OpenHTF is designed to reflect its core duties.
+If you're adding a new module or feature, put some thought into where it best
+fits in based on what it does.
+
+```
+  .-----------------.
+  |     openhtf     |
+  |-----------------|
+  | Python package. |
+  '-----------------'
+          |
+          |    .---------------------------------.
+          |    |              conf               |
+          |--->|---------------------------------|
+          |    | Reads and stores configuration. |
+          |    '---------------------------------'
+          |
+          |    .-------------------------.
+          |    |           exe           |
```

Comment:
hm, this may be better called 'core', i think that's a fairly common submodule for python packages to put things in that are "you probably shouldn't be touching this stuff from outside the library".